### PR TITLE
Add support for Google Tag Manager with dataLayer

### PR DESCRIPTION
**Pull request by salimhb**
See: https://developers.google.com/tag-manager/devguide

### CODE REVIEW

File: spec/integration/google_tag_manager_integration_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    setup_app(action: :google_tag_manager) do |tracker|
+      tracker.handler :google_tag_manager, { tracker: 'GTM-ABCDEF' }
+    end
+    visit '/'
+  end
+
+  subject { page }
+
+  it "embeds the script tag with tracking event from the controller action" do
+    expect(page.find("head")).to have_content('GTM-ABCDEF')
```
Comment (salimhb) [inline spec/integration/google_tag_manager_integration_spec.rb]: These are failing with jruby only. They work when dropping `find("head")` part.

Comment (kangguru) [inline spec/integration/google_tag_manager_integration_spec.rb]: `noscript` seems to be the issue, anyways google recommends to put the snippit in the body.

also see: http://www.lunametrics.com/blog/2014/12/12/google-tag-manager-snippet-placement/

Comment (salimhb) [inline spec/integration/google_tag_manager_integration_spec.rb]: That's absolutely right, it's even right there in the quickstart guide https://developers.google.com/tag-manager/quickstart
Thanks for catching this.

**Comment by jhilden** [comment]
looks good

**Comment by DonSchado** [comment]
can you directly add the change log entry and some documentation?

**Comment by salimhb** [comment]
@DonSchado thanks for the reminder. I updated the readme and changelog. And looking again at the GTM documentation, I change the main option from tracker to container which is the lingo used by GTM.

**Pull request merged**